templates/travis/.travis/before_install.sh.j2

```diff
@@ -68,27 +68,32 @@ if [ -n "$PULP_PR_NUMBER" ]; then
 fi
 {% endif %}
 
-git clone --depth=1 https://github.com/pulp/pulpcore-plugin.git
+# When building a (release) tag, we don't need the development modules for the
+# build (they will be installed as dependencies of the plugin).
+if [ -z "$TRAVIS_TAG" ]; then
 
-if [ -n "$PULP_PLUGIN_PR_NUMBER" ]; then
-  cd pulpcore-plugin
-  git fetch --depth=1 origin +refs/pull/$PULP_PLUGIN_PR_NUMBER/merge
-  git checkout FETCH_HEAD
-  cd ..
-fi
+  git clone --depth=1 https://github.com/pulp/pulpcore-plugin.git
 
+  if [ -n "$PULP_PLUGIN_PR_NUMBER" ]; then
+    cd pulpcore-plugin
+    git fetch --depth=1 origin +refs/pull/$PULP_PLUGIN_PR_NUMBER/merge
+    git checkout FETCH_HEAD
+    cd ..
+  fi
 
-git clone --depth=1 https://github.com/PulpQE/pulp-smash.git
 
-if [ -n "$PULP_SMASH_PR_NUMBER" ]; then
-  cd pulp-smash
-  git fetch --depth=1 origin +refs/pull/$PULP_SMASH_PR_NUMBER/merge
-  git checkout FETCH_HEAD
-  cd ..
-fi
+  git clone --depth=1 https://github.com/PulpQE/pulp-smash.git
```

Comment:
I agree with this design, but
1. Right now, the CI will fail with this I believe. Pulp-Smash's master branch has the container (kubectl) support. Pulp-Smash needs to do a release.
2. In general, we're going to need to do a pulp-smash release at the time we release pulpcore & pulpcore-plugin.